### ISSUE TITLE
Avoid duplicated sync on some situations

### ISSUE THREAD

**Issue by muojp** [muojp/sync-memories#6]
This issue sometimes happen on my device (Essential PH-1).

Repro:

1. insert USB stick
2. somehow nothing happened (probably due to high CPU load)
3. remove USB stick
4. insert USB stick again
5. wait for sync to be triggered

Expected: Files synched once and only once
Actual: Files synched twice .100% files are skipped so there's no redundant copy, but still consume much time for 100% useless checking.

**Comment by muojp** [comment]
There are roughly two solutions.

1. check for storage ID at the beginning of every sync and omit if the ID is the same as one before.

Pros: So fast.
Cons: Not complete. Somehow need to set timeout because the same storage would be used repeatedly and users will feel uncomfortable if intentional re-insertions are treated as nothing by the software. Therefore, showing menu on a notification bar (if a user is trying to sync the same storage within 5 minutes) and let the user decide re-sync or not would be a reasonable option.

2. Compare file list between sync sessions.

Pros: Accurate and hopefully fast enough if implemented correctly.
Cons: Time / space issue; keeping full file-list would be memory consuming. Let's say there would be 20k+ photos in a SD card which means 20MB+ memory needed if each entry needs 1KB of data. Extracting essential info (file name + file size would be the minimal) would reduce memory consumption a lot. And furthermore, we need to free memory at some moment if no longer needed (or expected to be unneeded).

**Comment by muojp** [comment]
Personally, I prefer the option (1) with 5 minutes timeout (auto-sync again after that period). It's simple, mostly go well without user interaction, and leave some room for user to make decision.

**Comment by muojp** [comment]
> there would be 20k+ photos in a SD card
This situation is not assumed on this app because due to file naming rule is DSC0????.JPG which means there would be a severe problem in case of 10k+ files available (kept); old files and new files are completely mixed dependent on their sizes (larger ones taken regardless of timestamp).

**Comment by muojp** [comment]
> there would be 20k+ photos in a SD card
This situation is not assumed on this app because due to file naming rule is DSC0????.JPG which means there would be a severe problem in case of 10k+ files available (kept); old files and new files are completely mixed dependent on their sizes (larger ones taken regardless of timestamp).
